File: src/main/kotlin/authserver/delta/report/Report.kt
package authserver.delta.report

import authserver.central.Central
import authserver.delta.assistance.Assistance
import authserver.delta.report.response.ReportResponse
import authserver.delta.worker.Worker
import authserver.utils.PaymentType
import jakarta.persistence.*
import java.util.*

@Entity
class Report (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var name: String = "",

    @Column(nullable = false)
    var description: String = "",

    @ManyToMany
    @JoinTable(
        name = "report_worker",
        joinColumns = [JoinColumn(name = "report_id")],
        inverseJoinColumns = [JoinColumn(name = "worker_id")]
    )
    var responsibleWorkers: MutableSet<Worker> = HashSet(),

    @Column
    var status: ReportStatus = ReportStatus.EM_ANALISE,

    @Column(nullable = false)
    var creationDate: Date,

    @Column
    var totalPrice: Float = 0.0f,

    @OneToOne(mappedBy = "report")
    var assistance: Assistance? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "Central_id")
    var responsibleCentral: Central,

    @Column
    var paymentType: PaymentType,

    @Column
    var machinePartExchange: Boolean,

    @Column
    var workDelayed: Boolean,

    ) {
    fun toResponse() = ReportResponse(id!!, name, description, creationDate.toString(), status.toString(), assistance?.id, responsibleWorkers.map { it.id!! }, totalPrice, paymentType, machinePartExchange, workDelayed)
}

File: front/Delta/lib/src/features/core/screens/report/report.dart
class ReportRequest {
  final String name;
  final String description;
  final List<num> responsibleWorkersIds;
  final num? assistanceId;
  final String paymentType;
  final bool machinePartExchange;
  final bool delayed;

  ReportRequest({
    required this.name,
    required this.description,
    required this.responsibleWorkersIds,
    required this.assistanceId,
    required this.paymentType,
    required this.machinePartExchange,
    required this.delayed
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'responsibleWorkersIds': responsibleWorkersIds,
      'assistanceId': assistanceId,
      'paymentType': paymentType,
      'machinePartExchange': machinePartExchange,
      'delayed': delayed
    };
  }
}


class ReportInformations {
  String? id;
  List<String> workersName;
  ReportResponse report;

  ReportInformations(this.id, this.workersName, this.report);

}

class UpdateReportRequest {
  final String name;
  final String description;
  final String status;
  final num? assistanceId;
  final List<num> responsibleWorkersIds;
  final String paymentType;
  final bool machinePartExchange;
  final bool delayed;

  UpdateReportRequest({required this.name,required this.description, required this.status,
    required this.assistanceId, required this.responsibleWorkersIds, required this.paymentType,
    required this.machinePartExchange, required this.delayed});

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'status': status,
      'assistanceId': assistanceId,
      'responsibleWorkersIds': responsibleWorkersIds,
      'paymentType': paymentType,
      'machinePartExchange': machinePartExchange,
      'delayed': delayed
    };
  }
}

class ReportResponse {
  final String? id;
  final String name;
  final String description;
  final String creationDate;
  final String status;
  final String? assistanceId;
  final List<String> responsibleWorkersIds;
  final String paymentType;
  final bool machinePartExchange;
  final bool delayed;

  ReportResponse({required this.id, required this.name,required this.description, required this.creationDate, required this.status,
  required this.assistanceId, required this.responsibleWorkersIds, required this.paymentType,
    required this.machinePartExchange, required this.delayed});

  Map<String, dynamic> toJson() {
    return {
      'id' : id,
      'name': name,
      'description': description,
      'creationDate': creationDate,
      'status': status,
      'assistanceId': assistanceId,
      'responsibleWorkersIds': responsibleWorkersIds,
      'paymentType': paymentType,
      'machinePartExchange': machinePartExchange,
      'delayed': delayed
    };
  }
}